File: codeBase/ACES_Coders/v8.0/pre-coders/Journey-on-tiles/Solution.py
'''
    This code was written by UOP_Bitlasagna during ACES Precoders v8 2020
    dinuransika98@gmail.com	sathirasofte@gmail.com	sasanka14@gmail.com

'''
n = int(input())
grid = [list(map(int, input().split())) for _ in range(n)]

for i in range(n - 2, -1, -1):
    grid[n - 1][i] += grid[n - 1][i + 1]
    grid[i][n - 1] += grid[i + 1][n - 1]
    
for i in range(n - 2, -1, -1):
    for j in range(n - 2, -1, -1):
        grid[i][j] += max(grid[i + 1][j], grid[i][j + 1])
        
print(grid[0][0])

File: codeBase/ACES_Coders/v8.0/pre-coders/Journey-on-tiles/Solution.java
/*
    This code was written by UOP_IntElligence during ACES Precoders v8 2020
    hisnionline@gmail.com   muhammadh.aslamm@gmail.com  irfanmm96@gmail.com
*/
import java.io.*;
import java.util.*;

public class Solution {
    
    static int max(int x, int y){ 
        return (x > y) ? x : y;  
    }
      
    private static int minCost(int cost[][], int m, int n){ 
        int i, j; 
        int tc[][]=new int[m+1][n+1]; 
  
        tc[0][0] = cost[0][0]; 
  
        for (i = 1; i <= m; i++) 
            tc[i][0] = tc[i-1][0] + cost[i][0]; 
  
        for (j = 1; j <= n; j++) 
            tc[0][j] = tc[0][j-1] + cost[0][j]; 
  
        for (i = 1; i <= m; i++) 
            for (j = 1; j <= n; j++) 
                tc[i][j] = max(tc[i-1][j], tc[i][j-1]) + cost[i][j]; 
  
        return tc[m][n]; 
    } 

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        int ROW = scanner.nextInt();
        int [][] arr = new int[ROW][ROW];
        
        for(int i=0;i<ROW;i++){
            for(int j=0;j<ROW;j++){
                arr[i][j] = scanner.nextInt();      
            }
        }    
        
        System.out.print( minCost(arr, ROW-1, ROW-1));
    }
}